chainer/training/extensions/value_observation.py

```diff
@@ -0,0 +1,46 @@
+import time
+
+from chainer.training import extension
+
+
+def observe_value(key, target_func):
+    """Returns a trainer extension to continuously record a value.
+
+    Args:
+        key (str): Key of observation to record.
+        target_func (function): Function that returns the value to record.
+            It must take one argument: :class:~chainer.training.Trainer object.
+    Returns:
+        The extension function.
+    """
+    @extension.make_extension(
+        trigger=(1, 'epoch'), priority=extension.PRIORITY_WRITER)
+    def _observe_value(trainer):
+        trainer.observation[key] = target_func(trainer)
+    return _observe_value
+
+
+def observe_time(key='time'):
+    """Returns a trainer extension to record the elapsed time.
+
+    Args:
+        key (str): Key of observation to record.
+
+    Returns:
+        The extension function.
+    """
+    start_time = time.time()
+    return observe_value(key, lambda _: time.time() - start_time)
+
+
+def observe_lr(optimizer, key='lr'):
```

Comment:
Optimizer can be extracted via trainer: `trainer.updater.get_optimizer('main')`. When user sets multiple optimizers to the updater, we should specify the key other than `'main'`. So how about accepting the optimizer name instead of the optimizer object itself, and making `'main'` the default?